### PR TITLE
doc: add quick-start guide

### PR DESCRIPTION
**Pull request by Madhu-1**
# Describe what this PR does #

added a quick start guide for the minimal installation
## Is there anything that requires special attention ##

List items that are not part of the PR and do not impact it's
functionality, but are work items that can be taken up subsequently.

**Checklist:**

* [ ] **Commit Message Formatting**: Commit titles and messages follow

  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi-operator/blob/devel/docs/development-guide.md#commit-messages).
* [ ] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi-operator/blob/devel/docs/development-guide.md#development-workflow)
* [ ] [Pending release
  notes](https://github.com/ceph/ceph-csi-operator/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [ ] Documentation has been updated, if necessary.
* [ ] Unit tests have been added, if necessary.
* [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by Madhu-1** [comment]
@nb-ohad PTAL

File: docs/quick-start.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+Create an OperatorConfig CR to configure the Ceph-CSI Operator:
+
+```console
+echo '
+apiVersion: csi.ceph.io/v1alpha1
+kind: OperatorConfig
+metadata:
+  name: ceph-csi-operator-config
+  namespace: ceph-csi-operator-system
+' | kubectl create -f -
```
Comment (nb-ohad) [inline docs/quick-start.md:49]: If we are going for minimal installation, why do we need an operator config instance?

File: docs/quick-start.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+rbd.csi.ceph.com-ctrlplugin-6965dcfdb8-w88kn            5/5     Running   0          4m35s
+rbd.csi.ceph.com-nodeplugin-lnm4n                       2/2     Running   0          4m35s
+```
+
+## 6. Clean Up Resources
+
+To clean up the resources, delete the OperatorConfig CR, drivers:
+
+```console
+kubectl delete operatorconfig ceph-csi-operator-config -n ceph-csi-operator-system
```
Comment (nb-ohad) [inline docs/quick-start.md:113]: Continued from the last comment, for a minimal installation example we don't need an operator config CR

File: docs/quick-start.md
Comment (nb-ohad) [inline docs/quick-start.md:1]: If we want a minimal installation that can actually serve storage, we need the user to create at least one ceph connection and at least one client profile. Both are missing from this quick start guide.

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: docs/quick-start.md (hunk 1 above)
Comment (Madhu-1) [reply docs/quick-start.md]: it was added to deploy the CR with minimal values and no values and the user can update it later as per requirement. Removed it now

**Comment by Madhu-1** [comment]
@nb-ohad PTAL

File: docs/quick-start.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+echo '
+apiVersion: csi.ceph.io/v1alpha1
+kind: ClientProfile
+metadata:
+  name: storage
+  namespace: ceph-csi-operator-system
+spec:
+  cephConnectionRef:
+    name: ceph-connection
+  cephFS:
```
Comment (iPraveenParihar) [inline docs/quick-start.md:126]:
```suggestion
  cephFs:
```

```
$ k explain clientprofiles.csi.ceph.io.spec.cephFs
GROUP:      csi.ceph.io
KIND:       ClientProfile
VERSION:    v1alpha1

FIELD: cephFs <Object>
```

**Review by obnoxxx — COMMENTED** [review]
I am surprised that the guide does not mention building the operator. Even if this is making use of published images, I think developers want to test locally  built images with their local changes.

**Comment by Madhu-1** [comment]
> I am surprised that the guide does not mention building the operator. Even if this is making use of published images, I think developers want to test locally built images with their local changes.

That will be a follow-up to add/update the document on how to test the local changes.

**Review by obnoxxx — COMMENTED** [review]
@Madhu-1  thanks for adding quickstart guide. Very useful!

A general suggestion:
Instead of the echo and using `kubectl create -f -`, wouldn't it be convenient to have example yaml files for these items (rbd driver, cephfs driver, client connection, client profile, etc)? this would probably be as follow-up PR.

**Review by obnoxxx — CHANGES_REQUESTED** [review]
I tried the steps in the guide and they all work with one exception:

Creation of the `clientProfile` fails like this:

```console
Error from server (BadRequest): error when creating "STDIN": ClientProfile in version "v1alpha1" cannot be handled as a ClientProfile: strict decoding error: unknown field "spec.cephFs.subvolumeGroup"
```

File: docs/quick-start.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  - [6. Create ClientProfile](#6-create-clientprofile)
+
+# Quick Start Guide for Ceph-CSI-Operator
+
+## 1. Prerequisites
+
+Before deploying the Ceph-CSI-Operator, ensure the following requirements are met:
+
+- A Kubernetes cluster ([supported version](https://kubernetes.io/releases/) recommended)
+- Ceph cluster ([supported version](https://docs.ceph.com/en/latest/releases/) recommended)
```
Comment (obnoxxx) [inline docs/quick-start.md:19]: Might be nice to mention the rook project here as a convenient means of creating a Ceph cluster in k8s.

File: docs/quick-start.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+```console
+echo '
+apiVersion: csi.ceph.io/v1alpha1
+kind: CephConnection
+metadata:
+  name: ceph-connection
+  namespace: ceph-csi-operator-system
+spec:
+  monitors:
+  - 10.98.44.171:6789
```
Comment (obnoxxx) [inline docs/quick-start.md:107]: It might be useful to explain how to get the requires IP address and port, e. g. when using rook to set up Ceph.

**Review by obnoxxx — CHANGES_REQUESTED** [review]
two comments  inline. Not strictly change requests really, rather suggestions/thoughts.

File: docs/quick-start.md (hunk 4 above)
Comment (Madhu-1) [reply docs/quick-start.md:19]: I have kept the options open for deploying Rook and its up to the user to use their own way to deploy the ceph cluster, that why i have mentioned that a supported version of ceph need to be installed.

File: docs/quick-start.md (hunk 5 above)
Comment (Madhu-1) [reply docs/quick-start.md:107]: As the ceph cluster is configured I assume that the user knows the basic monitor information to configure anything. that is the reason not pointing to any specific commands as it depends on the way of the ceph cluster deployment.

**Review by obnoxxx — APPROVED** [review]
LGTM

**Comment by Madhu-1** [comment]
@nb-ohad PTAL

File: docs/quick-start.md (hunk 5 above)
Comment (nb-ohad) [reply docs/quick-start.md:107]: @obnoxxx I agree with @Madhu-1 
Getting the IP address can be different based on how Ceph was deployed (rook or other means), we should not be based toward a specific deployment type and the assumption is that Ceph is already deployed.

**Review by nb-ohad — APPROVED** [review]

**Pull request merged**